systemtest/src/main/java/io/strimzi/systemtest/resources/operator/OlmResource.java

```diff
@@ -6,71 +6,260 @@
+            // if value is FALSE we are gonna use it = non-used install plan
+            if (!entry.getValue()) {
+                nonUsedInstallPlan[0] = entry.getKey();
+                break;
+            }
+            nonUsedInstallPlan[0] = NO_MORE_NON_USED_INSTALL_PLANS;
+        }
+
+        LOGGER.info("Non-used install plan is {}", nonUsedInstallPlan[0]);
+        return nonUsedInstallPlan[0];
+    }
+
+    /**
+     * Patches specific non used install plan, which will approve installation. Only for manual installation strategy.
+     * Also updates closedMapInstallPlan map and set specific install plan to true.
+     */
+    private static void approveNonUsedInstallPlan() {
+        String nonUsedInstallPlan = getNonUsedInstallPlan();
+
+        try {
+            LOGGER.info("Approving {} install plan", nonUsedInstallPlan);
+            String dynamicScriptContent =
+                "#!/bin/bash\n" +
+                    cmdKubeClient().cmd() +
+                    " patch installplan " + nonUsedInstallPlan + " --type json  --patch '[{\"op\": \"add\", \"path\": \"/spec/approved\", \"value\": true}]'";
+
+            InputStream inputStream = new ByteArrayInputStream(dynamicScriptContent.getBytes(Charset.defaultCharset()));
+            File patchScript = File.createTempFile("installplan_patch",  ".sh");
+            Files.copy(inputStream, patchScript.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+            Exec.exec("bash", patchScript.getAbsolutePath());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Upgrade cluster operator by obtaining new install plan, which was not used and also approves installation by
+     * changing the install plan YAML
+     */
+    public static void upgradeClusterOperator() {
+        if (kubeClient().listPodsByPrefixInName(Constants.CO_POD_PREFIX_NAME).get(0) == null) {
+            throw new RuntimeException("We can not perform upgrade! Cluster operator pod is not present.");
+        }
+
+        obtainInstallPlanName();
+        approveNonUsedInstallPlan();
+    }
+
+    /**
+     * Creates OperatorGroup from `olm/operator-group.yaml` and modify "${OPERATOR_NAMESPACE}" attribute in YAML
+     * @param namespace namespace where you want to apply OperatorGroup  kind
+     */
+    private static void createOperatorGroup(String namespace) {
+        try {
+            File operatorGroupFile = File.createTempFile("operatorgroup", ".yaml");
+            InputStream groupInputStream = OlmResource.class.getClassLoader().getResourceAsStream("olm/operator-group.yaml");
+            String operatorGroup = TestUtils.readResource(groupInputStream);
+            TestUtils.writeFile(operatorGroupFile.getAbsolutePath(), operatorGroup.replace("${OPERATOR_NAMESPACE}", namespace));
+            ResourceManager.cmdKubeClient().apply(operatorGroupFile);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Comment:
Why it's int try/catch block?

Reply:
`File.createTempFile("operatorgroup", ".yaml");` needs it.